ember-exclaim/declarations/components/exclaim-ui.d.ts

```diff
@@ -0,0 +1,67 @@
+import { ComponentLike } from '@glint/template';
+import { ComponentSpec, ImplementationMap } from '../index';
+
+declare const ExclaimUi: ComponentLike<{
+  Args: {
+    /**
+     * A spec for an Exclaim UI.
+     */
+    ui: unknown;
```

Comment:
Would it make sense to try to type this at all? At the very least, it would be `Record<string, unknown>`, right? I've attempted some more advanced recursive types for this in the past, but at this point not sure if something like that makes any sense.

Reply:
Hot take: imho `Record<string, unknown>` is worse than `unknown` for cases like this. It does rule out "fundamental misunderstanding" bugs like `@ui={{123}}`, but realistically those aren't the kinds of errors that are likely to be present in a UI config object. `Record<string, unknown>` suggests some degree of type safety at least at the top level where really none exists.

You can go super deep on deriving approximate valid types for a UI config based on an implementation map (particularly if you assume some domain-specific understanding of the contents of `meta`), but ultimately the reality is that if you actually statically know the type of your UI config, you don't have any reason to be using Exclaim and should probably invoke your components directly.

Reply:
Totally fair and good points. Ultimately, UI config is usually sourced from something dynamic and untyped, like JSON (otherwise, like you say, what's the point of using exclaim) so it makes sense that we should add no artificial type restriction here.